navigation/libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/navigation/NavigationRoute.java

```diff
@@ -50,17 +64,7 @@
   abstract String language();
 
   public void getRoute(@NonNull Callback<DirectionsResponse> callback) {
-    List<Callback<DirectionsResponse>> callbacks = new ArrayList<>(1);
-    callbacks.add(callback);
-    getRoute(callbacks);
-  }
-
-  public void getRoute(@NonNull List<Callback<DirectionsResponse>> callbacks) {
-
-    for (Callback<DirectionsResponse> callback : callbacks) {
-      // TODO am I making the request here multiple times?
-      getDirectionsRequest().enqueueCall(callback);
-    }
+    getDirectionsRequest().enqueueCall(callback);
```

Comment:
Would it be possible to create your own callback that then calls the original callback? And then extract the profile from the builder here to set in the DirectionsResponse?